specification/communication/data-plane/Microsoft.CommunicationServicesIdentity/preview/2021-03-31-preview/CommunicationIdentity.json

```diff
@@ -0,0 +1,118 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "CommunicationIdentityClient",
+    "description": "Azure Communication Identity Service",
+    "version": "2021-03-31-preview"
+  },
+  "paths": {
+    "/m365identities/:issueAccessToken": {
+      "post": {
+        "tags": [
+          "TeamsToken"
+        ],
+        "summary": "Generate a new M365 Skypetoken in exchange for ACS AAD token for an identity.",
```

Comment:
We are using many terms here: M365, Teams, Skype, ACS and AAD. What is the way that we want developers undrestand this. Is it accurate to say this route generates a communication token for an Azure Active Directory user? If so, let's use that language, if not maybe a more clear description.

Reply:
@DominikMe what do you think?

Reply:
I think the terminology here is Azure Active Directory as token input and Azure Communication Service as token output. Does the AAD token need to have specific Teams scopes defined in order for the request to succeed?

Reply:
Another proposal can be "Issue a new M365 access token in exchange of AAD access token."

Reply:
AFAIK we need to add scopes as "Voip" and "Chat". But I need to sync with colleagues what will happen if we use only one of them or non of them.

Reply:
We need at least one of them

Reply:
> Another proposal can be "Issue a new M365 access token in exchange of AAD access token."
But is that what we are doing? If I understood correctly, this will receive AAD token and returns an ACS token. Am I mistaken?

Reply:
> We need at least one of them

Do you mean the AAD token needs to have at least one of those scopes or do you mean we need a scope parameter for the route?

Reply:
these scopes are part of AAD token. 
Regarding descirption: @RezaJooyandeh you are right, the result is token for ACS. It should be the same token as for issuingToken(the same as for GA). there we use this description: "Issue a new token for an identity." , So maybe we can change this to "Issue a new token in exchange of AAD access token."

Reply:
example of ftpToken 
![image](https://user-images.githubusercontent.com/17181592/116106194-4cee3080-a6b2-11eb-84cc-18f914654226.png)

Reply:
So does it mean that in the token exchange process, we check what scopes existed in the original token and then give them a token with the corresponding ACS scopes?

Reply:
1st party app control these scopes, but in ACS AUTH service, we do not control it for private preview.
Here you have an example of the response from the endpoint. 
![image](https://user-images.githubusercontent.com/17181592/116222303-c5a2ca80-a74e-11eb-83b5-21b464fa69f6.png)
where 780 is bit flag using these values https://skype.visualstudio.com/DefaultCollection/SCC/_git/service-shared_framework_skypetoken?path=%2Fsrc%2FAbstractions%2FSkypetokenScope.cs. You can see our specification here https://microsoft.sharepoint-df.com/:w:/t/IC3SDK/EXw-h08WdTpGu7puT_wJWNgBK6WuFR6N-C6cQaU8eR2Uhg